kv/txn_correctness_test.go

```diff
@@ -178,28 +183,34 @@ func scanCmd(c *cmd, txn *client.Txn, t *testing.T) *roachpb.Error {
 	return nil
 }
 
-// incCmd adds one to the value of c.key in the env and writes
-// it to the db. If c.key isn't in the db, writes 1.
+// incCmd adds one to the value of c.key in the env (as determined by
+// a previous read, or else assumed to be zero) and writes it to the
+// db. If c.key isn't in the db, writes 1.
 func incCmd(c *cmd, txn *client.Txn, t *testing.T) *roachpb.Error {
-	r, pErr := txn.Inc(c.getKey(), 1)
-	if pErr != nil {
+	r := c.env[c.key] + 1
+	if pErr := txn.Put(c.getKey(), r); pErr != nil {
 		return pErr
 	}
-	c.env[c.key] = r.ValueInt()
-	c.debug = fmt.Sprintf("[%d]", r.ValueInt())
+	c.env[c.key] = r
+	c.debug = fmt.Sprintf("[%d]", r)
 	return nil
 }
 
-// sumCmd sums the values of all keys != c.key read during the transaction and
```

Comment:
This sum command was really confusing and the new write replacement is a lot more obvious (though the parsing and everything here is really unsophisticated).